custom_components/waste_collection_schedule/waste_collection_schedule/source/peterborough_gov_uk.py

```diff
@@ -125,6 +132,9 @@ def fetch(self):
 
         # get schedule
         schedule = self.get_schedule(s, self._post_code, self._uprn)
+        if schedule is None:
+            _LOGGER.error(f"No schedule data retrieved for postcode {self._post_code}, UPRN {self._uprn}")
+            return []  # Return empty list to avoid crashing HA
```

Comment:
> log an error and return an empty list

Raising an Exception does not crash anything but produce a log https://github.com/mampfes/hacs_waste_collection_schedule/blob/994a86c511d1b8c4afe1cd5a2823cd38b4dff15b/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py#L154-L163

I prefer raising an Exception as generates uniform error messages and contains a strack trace in the logs